### PR TITLE
Add safe sprig functions to template engine

### PR DESCRIPTION
**Pull request by kostola**
Enrich template engine with [sprig functions](http://masterminds.github.io/sprig/).

There is a filter phase that requires every function to be explicitly opted-in

This avoids to include by mistake some functions that may cause security threats (like access to OS environment variables, filesystem and network).

### CODE REVIEW

File: internal/packages/packageloader/transformfiles_funcs.go [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+	"mustSlice":          {},
+	"concat":             {},
+	"dig":                {},
+	"chunk":              {},
+	"mustChunk":          {},
+
+	// Crypto
+	"bcrypt":                   {},
+	"htpasswd":                 {},
+	"genPrivateKey":            {},
```
Comment (thetechnick) [inline internal/packages/packageloader/transformfiles_funcs.go:207]: I don't think generating private keys is independent

File: internal/packages/packageloader/transformfiles_funcs.go [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+	"buildCustomCert":          {},
+	"genCA":                    {},
+	"genCAWithKey":             {},
+	"genSelfSignedCert":        {},
+	"genSelfSignedCertWithKey": {},
+	"genSignedCert":            {},
+	"genSignedCertWithKey":     {},
+	"encryptAES":               {},
+	"decryptAES":               {},
+	"randBytes":                {},
```
Comment (thetechnick) [inline internal/packages/packageloader/transformfiles_funcs.go:218]: A lot of these functions are also producing randomized outputs.
Creating and working with certificates within our template process sounds like a really bad idea, I would like to leave those out as well.

File: internal/packages/packageloader/transformfiles_funcs.go [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+
+	// UUIDs
+	"uuidv4": {},
```
Comment (thetechnick) [inline internal/packages/packageloader/transformfiles_funcs.go:221]: Randomly generated UUIDs can also not be part of the template or they are changing the output on every run :)

File: internal/packages/packageloader/transformfiles_funcs.go [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+	"repeat":       {},
+	"trimAll":      {},
+	"trimSuffix":   {},
+	"trimPrefix":   {},
+	"nospace":      {},
+	"initials":     {},
+	"randAlphaNum": {},
+	"randAlpha":    {},
+	"randAscii":    {},
+	"randNumeric":  {},
```
Comment (thetechnick) [inline internal/packages/packageloader/transformfiles_funcs.go:50]: no randomization

File: internal/packages/packageloader/transformfiles_funcs.go [hunk 5]
```diff
@@ -0,0 +1,258 @@
...
+	"swapcase":     {},
+	"shuffle":      {},
```
Comment (thetechnick) [inline internal/packages/packageloader/transformfiles_funcs.go:52]: also random

File: internal/packages/packageloader/transformfiles_funcs.go [hunk 6]
```diff
@@ -0,0 +1,258 @@
...
+	"duration":         {},
+	"durationRound":    {},
+	"htmlDate":         {},
+	"htmlDateInZone":   {},
+	"must_date_modify": {},
+	"mustDateModify":   {},
+	"mustToDate":       {},
+	"now":              {},
+	"toDate":           {},
+	"unixEpoch":        {},
```
Comment (thetechnick) [inline internal/packages/packageloader/transformfiles_funcs.go:29]: Please drop `now` `unixEpoch` `html*` `ago`  and what's with the lower snake cased function names?
Is it `date_in_zone` or `dateInZone`? The documentation only mentions the latter.

**Review by thetechnick — CHANGES_REQUESTED** [review]

File: internal/packages/packageloader/transformfiles_funcs.go (hunk 1 above)
Comment (eqrx) [reply internal/packages/packageloader/transformfiles_funcs.go:207]: *important? or what do you mean by independent?:)

File: internal/packages/packageloader/transformfiles_funcs.go (hunk 2 above)
Comment (eqrx) [reply internal/packages/packageloader/transformfiles_funcs.go:218]: :+1:

File: internal/packages/packageloader/transformfiles_funcs.go (hunk 5 above)
Comment (eqrx) [reply internal/packages/packageloader/transformfiles_funcs.go:52]: yes please kill all of them. Not totally related but https://reproducible-builds.org/ is a nice source (in my opinion) why that matters in general, just FYI :)

File: internal/packages/packageloader/transformfiles_funcs.go (hunk 1 above)
Comment (thetechnick) [reply internal/packages/packageloader/transformfiles_funcs.go:207]: *idempotent, sorry :D

**Comment by kostola** [comment]
@thetechnick @eqrx I removed all the dates, random, crypto, os, network and filepath functions. It seems safe now for me.

File: internal/packages/packageloader/transformfiles_funcs.go (hunk 1 above)
Comment (kostola) [reply internal/packages/packageloader/transformfiles_funcs.go]: see [here](https://github.com/package-operator/package-operator/pull/172#issuecomment-1431244223)

File: internal/packages/packageloader/transformfiles_funcs.go (hunk 2 above)
Comment (kostola) [reply internal/packages/packageloader/transformfiles_funcs.go]: see [here](https://github.com/package-operator/package-operator/pull/172#issuecomment-1431244223)

File: internal/packages/packageloader/transformfiles_funcs.go (hunk 4 above)
Comment (kostola) [reply internal/packages/packageloader/transformfiles_funcs.go]: see [here](https://github.com/package-operator/package-operator/pull/172#issuecomment-1431244223)

File: internal/packages/packageloader/transformfiles_funcs.go (hunk 5 above)
Comment (kostola) [reply internal/packages/packageloader/transformfiles_funcs.go]: see [here](https://github.com/package-operator/package-operator/pull/172#issuecomment-1431244223)

File: internal/packages/packageloader/transformfiles_funcs.go (hunk 6 above)
Comment (kostola) [reply internal/packages/packageloader/transformfiles_funcs.go]: see [here](https://github.com/package-operator/package-operator/pull/172#issuecomment-1431244223)

**Review by eqrx — APPROVED** [review]
Needs docs but code is ok :)

File: internal/packages/packageloader/transformfiles_funcs.go (hunk 3 above)
Comment (kostola) [reply internal/packages/packageloader/transformfiles_funcs.go]: see [here](https://github.com/package-operator/package-operator/pull/172#issuecomment-1431244223)

**Pull request merged**